identity_presentation/src/commonMain/kotlin/net/thechance/mena/identity/presentation/screen/addresses/addEditLocation/SaveAddressStrategyFactory.kt

```diff
@@ -0,0 +1,15 @@
+package net.thechance.mena.identity.presentation.screen.addresses.addEditLocation
+
+import kotlin.uuid.ExperimentalUuidApi
+import kotlin.uuid.Uuid
+
+class SaveAddressStrategyFactory {
+    @OptIn(ExperimentalUuidApi::class)
+    fun createStrategy(addressId: Uuid?): ISaveAddress {
+        return if (addressId == null) {
+            CreateAddress()
+        } else {
+            UpdateAddress()
+        }
+    }
+}
```

Comment:
This is not really a factory should be an object or inject strategies via constructor
```suggestion
     class AddressOperationStrategyFactory(
         private val createStrategy: CreateAddressStrategy,
         private val updateStrategy: UpdateAddressStrategy
     ) {
         fun getStrategy(addressId: Uuid?): AddressOperationStrategy {
             return if (addressId == null) createStrategy else updateStrategy
         }
     }
```